File: src/main/java/ordilov/lissn/track/infrastructure/TrackRepository.java
package ordilov.lissn.track.infrastructure;

import java.util.Optional;
import ordilov.lissn.track.domain.Track;
import org.springframework.data.jpa.repository.JpaRepository;

public interface TrackRepository extends JpaRepository<Track, Long> {
  Optional<Track> findByResourceId(String resourceId);
}

File: src/main/java/ordilov/lissn/track/infrastructure/TrackRepository.kt
package ordilov.lissn.track.infrastructure;

import ordilov.lissn.track.domain.Track
import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface TrackRepository : JpaRepository<Track, Long> {
    fun findByResourceId(resourceId: String): Optional<Track>
}